homeassistant/components/heatzy/climate.py

```diff
@@ -0,0 +1,294 @@
+            await self.coordinator.async_request_refresh()
+        except HeatzyException as error:
+            _LOGGER.error(
+                "Set preset mode (%s) %s (%s)",
+                preset_mode,
+                error,
+                self.name,
+                exc_info=True,
+            )
+
+
+class HeatzyPiloteV2Thermostat(HeatzyThermostat):
+    """Heaty Pilote v2."""
+
+    HEATZY_TO_HA_STATE = {
+        "cft": PRESET_COMFORT,
+        "eco": PRESET_ECO,
+        "fro": PRESET_AWAY,
+        "stop": PRESET_NONE,
+    }
+
+    HA_TO_HEATZY_STATE = {
+        PRESET_COMFORT: "cft",
+        PRESET_ECO: "eco",
+        PRESET_AWAY: "fro",
+        PRESET_NONE: "stop",
+    }
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return self.HEATZY_TO_HA_STATE.get(self._conf_attr.get(CONF_MODE))
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        try:
+            await self.coordinator.api.async_control_device(
+                self.unique_id,
+                {CONF_ATTRS: {CONF_MODE: self.HA_TO_HEATZY_STATE.get(preset_mode)}},
+            )
+            await self.coordinator.async_request_refresh()
+        except HeatzyException as error:
+            _LOGGER.error(
+                "Set preset mode (%s) %s (%s)",
+                preset_mode,
+                error,
+                self.name,
+                exc_info=True,
+            )
+
+
+class Glowv1Thermostat(HeatzyPiloteV2Thermostat):
+    """Heatzy Glow."""
+
+    @property
+    def supported_features(self) -> int:
+        """Return supported features."""
+        return (
+            ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            | ClimateEntityFeature.PRESET_MODE
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return current temperature."""
+        cur_temp_high = self._conf_attr.get(CUR_TEMP_H, 0)
+        cur_temp_low = self._conf_attr.get(CUR_TEMP_L, 0)
+        return (cur_temp_low + (cur_temp_high * 255)) / 10
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Return comfort temperature."""
+        cft_temp_high = self._conf_attr.get(CFT_TEMP_H, 0)
+        cft_temp_low = self._conf_attr.get(CFT_TEMP_L, 0)
+        return (cft_temp_low + (cft_temp_high * 255)) / 10
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Return comfort temperature."""
+        eco_temp_high = self._conf_attr.get(ECO_TEMP_H, 0)
+        eco_temp_low = self._conf_attr.get(ECO_TEMP_L, 0)
+        return (eco_temp_low + (eco_temp_high * 255)) / 10
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temp_eco = kwargs[ATTR_TARGET_TEMP_LOW]
+        temp_cft = kwargs[ATTR_TARGET_TEMP_HIGH]
+
+        if (temp_eco or temp_cft) is None:
+            return
```

Comment:
I don't think this will ever be true – did you mean
```suggestion
        if temp_eco is None or temp_cft is None:
            return
```